packages/common-web/src/arrays.ts

```diff
@@ -1,11 +1,8 @@
-export const keyBy = <T>(arr: T[], key: string): Record<string, T> => {
-  return arr.reduce(
-    (acc, cur) => {
-      acc[cur[key]] = cur
-      return acc
-    },
-    {} as Record<string, T>,
-  )
+export const keyBy = <T>(arr: T[], key: string): Map<string, T> => {
```

Comment:
Note that this is a breaking change, requiring a `minor` version update to the changeset.